ricecooker/classes/files.py

```diff
@@ -551,10 +552,14 @@ class AudioFile(DownloadFile):
     default_ext = file_formats.MP3
     allowed_formats = [file_formats.MP3]
     is_primary = True
+    duration = None
 
     def get_preset(self):
         return self.preset or format_presets.AUDIO
 
+    def get_duration(self):
```

Comment:
I think it might be simpler to not add the `get_duration` methods, and just add `process_file` to the `AudioFile` class. As it derives from `DownloadFile` can call super on the `process_file` method, then set the duration on the returned filename.

Reply:
Changed this too